index.js

```diff
@@ -83,13 +83,23 @@ const npmWalker = Class => class Walker extends Class {
     opt.path = opt.path || process.cwd()
     const dirName = path.basename(opt.path)
     const parentName = path.basename(path.dirname(opt.path))
-    opt.follow =
-      dirName === 'node_modules' ||
-      (parentName === 'node_modules' && /^@/.test(dirName))
+
+    // only follow links in the root node_modules folder, because if those
+    // folders are included, it's because they're bundled, and bundles
+    // should include the contents, not the symlinks themselves.
+    // This regexp tests to see that we're either a node_modules folder,
+    // or a @scope within a node_modules folder, in the root's node_modules
+    // hierarchy (ie, not in test/foo/node_modules/ or something).
+    const followRe = /^(?:\/node_modules\/(?:@[^\/]+\/[^\/]+|[^\/]+)\/)*\/node_modules(?:\/@[^\/]+)?$/
```

Comment:
woof, what a regex

Reply:
People are so afraid of regexes.  Maybe I've just used them too much, but this is entirely sensible to me 🤣

Reply:
It's just "any amount of `node_modules/$pkg` nesting, and then either a `node_modules` folder or a `@scope` folder".